controllers/awsmachine_controller.go

```diff
@@ -321,6 +322,22 @@ func (r *AWSMachineReconciler) reconcileNormal(ctx context.Context, machineScope
 	// TODO(vincepri): Remove this annotation when clusterctl is no longer relevant.
 	machineScope.SetAnnotation("cluster-api-provider-aws", "true")
 
+	// Set addresses for awsmachine status
+	addrs := []v1.NodeAddress{
+		v1.NodeAddress{
+			Type:    v1.NodeInternalIP,
+			Address: aws.StringValue(instance.PrivateIP),
+		},
+	}
+	if instance.PublicIP != nil {
+		publicAddress := v1.NodeAddress{
+			Type:    v1.NodeExternalIP,
+			Address: aws.StringValue(instance.PublicIP),
+		}
+		addrs = append(addrs, publicAddress)
+	}
+	machineScope.SetAddresses(addrs)
```

Comment:
Should this also potentially iterate over attached ENIs as well (we do allow users to specify one or more ENIs through `Machine.Spec.NetworkInterfaces` today)?